### ISSUE TITLE
BundleValidation doesn't detect changes correctly

### ISSUE THREAD

**Issue by KremnevDmitry** [vaadin/flow#17998]
### Description of the bug

If I add shadow DOM styles to my application, `BundleValidationUtil` will always mark it as a change.

![image](https://github.com/vaadin/flow/assets/75516667/8f983c69-c318-4b9f-9042-026b0a30679f)

Thus, the development bundle is created every time, which significantly slows down the application start.

### Expected behavior

Since there were no changes in theme components CSS files the bundle will not be rebuilt.

### Minimal reproducible example

See [test.zip](https://github.com/vaadin/flow/files/13279677/test.zip)

I took the base spring boot vaadin application and added the shadow DOM styles: `frontend/themes/my-theme/components/vaadin-form-layout.css`.

Every time the application is started, the logs contain information about changes:
<details>
  <summary>LOG</summary>
  
```
2023-11-07T16:10:22.058+04:00  INFO 3532 --- [  restartedMain] c.v.f.s.frontend.BundleValidationUtil    : Checking if a development mode bundle build is needed
2023-11-07T16:10:22.378+04:00  INFO 3532 --- [  restartedMain] c.v.f.s.frontend.BundleValidationUtil    : Detected new or changed theme components CSS files
  themes\my-theme\components\vaadin-form-layout.css

2023-11-07T16:10:22.378+04:00  INFO 3532 --- [  restartedMain] c.v.f.s.frontend.BundleValidationUtil    : Detected removed theme components CSS files
  themes/my-theme/components/vaadin-form-layout.css

2023-11-07T16:10:22.380+04:00  INFO 3532 --- [  restartedMain] c.v.f.s.frontend.BundleValidationUtil    : A development mode bundle build is needed
2023-11-07T16:10:22.501+04:00  INFO 3532 --- [onPool-worker-1] c.v.f.s.frontend.TaskUpdatePackages      : Running `npm install` to resolve and optionally download frontend dependencies. This may take a moment, please stand by...
2023-11-07T16:10:37.184+04:00  INFO 3532 --- [onPool-worker-1] c.v.f.s.frontend.TaskUpdatePackages      : using 'C:\Program Files\nodejs\npm.cmd --no-update-notifier --no-audit --scripts-prepend-node-path=true --ignore-scripts install' for frontend package installation
2023-11-07T16:11:20.550+04:00  INFO 3532 --- [onPool-worker-1] c.v.f.s.frontend.TaskUpdatePackages      : Frontend dependencies resolved successfully.
2023-11-07T16:11:26.173+04:00  INFO 3532 --- [onPool-worker-1] c.v.f.s.frontend.TaskCopyFrontendFiles   : Copying frontend resources from jar files
2023-11-07T16:11:26.436+04:00  INFO 3532 --- [onPool-worker-1] c.v.f.s.frontend.TaskCopyFrontendFiles   : Visited 20 resources. Took 262 ms.
2023-11-07T16:11:26.760+04:00  INFO 3532 --- [onPool-worker-1] c.v.f.s.frontend.TaskRunDevBundleBuild   : Creating a new development mode bundle. This can take a while but will only run when the project setup is changed, addons are added or frontend files are modified
2023-11-07T16:11:51.355+04:00  INFO 3532 --- [onPool-worker-1] c.v.f.s.frontend.TaskRunDevBundleBuild   : Development frontend bundle built
2023-11-07T16:11:51.386+04:00  WARN 3532 --- [onPool-worker-1] c.v.f.s.frontend.TaskRunDevBundleBuild   : Failed to create a README file in src/main/dev-bundle
```

</details>

### Versions

Flow: 24.2.2
Vaadin: 24.2.2
Java: Amazon.com Inc. 19.0.2
OS: amd64 Windows 11 10.0
Browser: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36

**Comment by caalador** [comment]
I can see from the log that it is not so much a change in the theme, but a path issue.
ThemeValidationUtil doesn't take into account windows and the usage of \ instead of /

**Comment by KremnevDmitry** [comment]
It makes sense. In which release can we expect a fix?

**Comment by caalador** [comment]
The fix will come out most likely with Vaadin 24.2.3  (at least in Flow 24.2.3)

**Comment by mshabarov** [comment]
Closed as pull request is in.
